pkg/scheduler/preemption/preemption.go

```diff
@@ -162,8 +162,20 @@ func canBorrowWithinCohort(cq *cache.ClusterQueue, wl *kueue.Workload) (bool, *i
 	return true, &threshold
 }
 
+const (
+	// ClusterQueueOrigin indicates that preemption originated from cluster queue
+	ClusterQueueOrigin = "ClusterQueue"
+	// CohortOrigin indicates that preemption originated from cohort
+	CohortOrigin = "cohort"
+
+	InClusterQueueReason = "InClusterQueue"
+
+	InCohortReclamationReason = "InCohortReclamation"
+	InCohortFairSharingReason = "InCohortFairSharing"
+)
+
 // IssuePreemptions marks the target workloads as evicted.
-func (p *Preemptor) IssuePreemptions(ctx context.Context, preemptor *workload.Info, targets []*workload.Info, cq *cache.ClusterQueue) (int, error) {
+func (p *Preemptor) IssuePreemptions(ctx context.Context, preemptor *workload.Info, targets []*workload.Info, cq *cache.ClusterQueue, reclamation bool) (int, error) {
```

Comment:
I prefer borrowing to reclaimation, as it makes the interface a little easier for the caller to use - the caller doesn't have to know, or consider what reclaimation (!borrowing) is -  just whether or not the preempting workload is borrowing